### PR TITLE
Add reference to README to command `--help`

### LINKED ISSUE #118: Please document install methods
The installer supports a wide variety of installation methods. Here is the list for `git-annex`:

- auto
- apt
- brew
- neurodebian
- deb-url
- autobuild
- snapshot
- conda
- datalad/git-annex:tested
- datalad/git-annex
- datalad/git-annex:release
- datalad/packages
- dmg

It would be very useful, if the documentation could say a few words about the differences and why one would choose one over the other. From the labels alone it is not clear how `autobuild` is different from `snapshot`, or `datalad/git-annex` from `datalad/git-annex:release` or `datalad/git-annex:tested`. Does `auto` pick the method that gives the latest on any platform?

Thx!

### PR DESCRIPTION
**Pull request by jwodder**
This PR extends the description shown when `datalad-installer --help` is run to include (among other things) instructions to consult the README in this repository for full documentation.

Closes #118.

### CODE REVIEW

**Comment by mih** [comment]
Thx!

File: src/datalad_installer.py [hunk 1]
```diff
@@ -471,7 +475,15 @@ class DataladInstaller:
...
     OPTION_PARSER = OptionParser(
-        help="Installation script for Datalad and related components",
+        help=(
+            "Installation script for Datalad and related components\n\n"
+            "`datalad-installer` is a script for installing Datalad, git-annex,"
+            " and related components all in a single invocation.  It requires"
+            " no third-party Python libraries, though it does make heavy use of"
+            " external packaging commands.\n\n"
+            "See the README at <https://github.com/datalad/datalad-installer>"
+            " for a complete descriptions of all options."
```
Comment (yarikoptic) [inline src/datalad_installer.py:485]:
```suggestion
            " for a complete description of all options."
```
or
```suggestion
            " for complete descriptions of all options."
```
?

File: test/test_options.py [hunk 2]
```diff
@@ -308,6 +308,14 @@ def test_global_long_help() -> None:
         "\n"
         "  Installation script for Datalad and related components\n"
         "\n"
+        "  `datalad-installer` is a script for installing Datalad, git-annex, and\n"
+        "  related components all in a single invocation.  It requires no third-party\n"
+        "  Python libraries, though it does make heavy use of external packaging\n"
+        "  commands.\n"
+        "\n"
+        "  See the README at <https://github.com/datalad/datalad-installer> for a\n"
```
Comment (yarikoptic) [inline test/test_options.py:316]: would need to be tuned up here if `a` is removed per above

File: src/datalad_installer.py (hunk 1 above)
Comment (jwodder) [reply src/datalad_installer.py]: Fixed.

**Comment by yarikoptic** [comment]
Thank you @jwodder !

**Pull request merged**